File: app/src/main/java/com/android/sonu/ummsplash/bus/AppBusModule.java
package com.android.sonu.ummsplash.bus;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * Created by amanshuraikwar on 18/12/17.
 */

@Module
public class AppBusModule {

    @Singleton
    @Provides
    static AppBus getAppBus() {
        return new AppBus();
    }
}

File: app/src/main/java/io/github/amanshuraikwar/splash/bus/AppBusModule.kt
package io.github.amanshuraikwar.splash.bus

import dagger.Module
import dagger.Provides
import javax.inject.Singleton

/**
 * Dagger Module to provide Event Bus related instances.
 *
 * @author Amanshu Raikwar
 * Created by amanshuraikwar on 07/03/18.
 */
@Module class AppBusModule {

    @Singleton @Provides fun appBus(): AppBus = AppBus()
}